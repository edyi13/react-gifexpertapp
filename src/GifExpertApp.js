import {useState} from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export  const GifExpertApp =  ()=>{
    // const categories = ['One Punch', 'Samurai X', 'Dragon Ball Z'];
    const [categories, setCategories] = useState (['One Piece']);

    // const handleAdd = (e) =>{
    //     //setCategories([...categories, 'One Punch']);
    //     setCategories(cats => [...categories, 'One Punch']);
    // }
    return(
      <>
        <h2>Segundo Componente React</h2>
        <AddCategory setCategories = { setCategories }/>
        <hr/>
       
        <ol>
           { 
                categories.map( category => (
                     //<li key= {category}>{category}</li>
                    <GifGrid key = { category } category = { category }/>
                ))
           }
        </ol>
      </>
    );
  };