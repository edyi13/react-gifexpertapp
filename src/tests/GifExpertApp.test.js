import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";

describe('Pruebas en el <GifExpertApp/>', () => {
    test('Deberia de hacer match con el snapshot', () => {
        //Assert
        const wrapper = shallow(<GifExpertApp/>);
        //Act
        //Assert
        expect(wrapper).toMatchSnapshot();
    })

    test('Deberia de mostrar una lista de categorias', () => {
        //Arrange
        const categories = ['One Piece','Jujutsu kaizen'];
        const wrapper = shallow(<GifExpertApp defaultCategories = { categories } />);
        //Act
        //Assert
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    })
    
    
})
