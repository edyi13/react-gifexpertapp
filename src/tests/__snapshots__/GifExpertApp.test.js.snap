// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifExpertApp/> Deberia de hacer match con el snapshot 1`] = `
<Fragment>
  <h2>
    Segundo Componente React
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en el <GifExpertApp/> Deberia de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    Segundo Componente React
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Piece"
      key="One Piece"
    />
    <GifGrid
      category="Jujutsu kaizen"
      key="Jujutsu kaizen"
    />
  </ol>
</Fragment>
`;
